spackbot/handlers/pipelines.py

```diff
@@ -52,20 +52,43 @@ async def run_pipeline(event, gh):
     branch = pr["head"]["ref"]
     branch = f"github/pr{number}_{branch}"
     branch = urllib.parse.quote_plus(branch)
-
-    url = f"{helpers.gitlab_spack_project_url}/pipeline?ref={branch}"
-    headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
-
-    logger.info(f"{sender} triggering pipeline, url = {url}")
-
-    # Don't provide GitHub credentials to GitLab!
-    async with aiohttp.ClientSession() as session:
-        async with session.post(url, headers=headers) as response:
-            result = await response.json()
-
+    if not retry_pipeline:
+        trigger_type = "started"
+        url = f"{helpers.gitlab_spack_project_url}/pipeline?ref={branch}"
+        headers = {"PRIVATE-TOKEN": GITLAB_TOKEN}
+
+        # Don't provide GitHub credentials to GitLab!
+        async with aiohttp.ClientSession() as session:
+            async with session.post(url, headers=headers) as response:
+                result = await response.json()
+    else:
+        # Get a list of failed pipelines
+        trigger_type = "retried"
+        query_string = f"?status=failed&order_by=updated_at&ref={branch}"
```

Comment:
Is it only `failed` pipelines we are looking to retry?  What if the pipeline we want to retry is one that was `canceled` or `skipped`?

Reply:
I hadn't thought about those.  I can look to see if we can filter out the passing ones and grab the latest of that list of pipelines